api/provision/mock-invocation-lambda.test.ts

```diff
@@ -0,0 +1,93 @@
+import { Context } from "aws-lambda";
+import { handler } from "./mock-invocation-lambda";
+import { CloudServiceProvider, Network } from "../../models/cloud-service-providers";
+import { ProvisionRequestType } from "../../models/provisioning-jobs";
+import { OtherErrorResponse, ValidationErrorResponse } from "../../utils/response";
+import { transformProvisionJob } from "./start-provision-job";
+
+const fundingSources = [
+  {
+    taskOrderNumber: "1234567890123",
+    clin: "9999",
+    popStartDate: "2021-07-01",
+    popEndDate: "2022-07-01",
+  },
+];
+const operators = ["admin1@mail.mil", "superAdmin@mail.mil"];
+const provisioningBodyNoPayload = {
+  jobId: "81b31a89-e3e5-46ee-acfe-75436bd14577",
+  userId: "21d18790-bf3e-4529-a361-460ee6d16e0b",
+  portfolioId: "b02e77d1-234d-4e3d-bc85-b57ca5a93952",
+  operationType: ProvisionRequestType.ADD_OPERATORS,
+  targetCsp: CloudServiceProvider.CSP_A.name,
+  targetNetwork: Network.NETWORK_1,
+};
+
+describe("Successful invocation of mock CSP function", () => {
+  it("should return 200 when CSP A provided in the request", async () => {
+    const stateInput = transformProvisionJob({
+      ...provisioningBodyNoPayload,
+      targetCsp: "CSP_A",
+      payload: {
+        name: "Mock A Invocation Portfolio",
+        fundingSources,
+        operators,
+      },
+    }) as any;
```

Comment:
We probably really do want to avoid all this `any`.

Reply:
Yeah, I can't argue with that.

Reply:
@kylelaker-ccpo I've removed the amount of `any` being used. The `any` being used by the responses was causing typescript to complain due to one of the types in the union not having a specific property even though one of the types have the property. Didn't think that was a problem, but maybe I am misunderstanding something. 
```typescript
Property 'code' does not exist on type 'CspResponse | ValidationErrorResponse'.
  Property 'code' does not exist on type 'ValidationErrorResponse'.
```

Reply:
Yeah, so the property has to exist on _all_ the possible types. `any` kinda bypasses those checks (basically, a way to say, "no `tsc`, I _do_ know what I'm doing). So of `CspResonse` and `ValidationErrorResponse`, that error implies that `ValidationErrorResponse` doesn't have a `code` property; so we'd get either `undefined` or an error if we tried to access it.

So we'd have to do at least one of the following:
 - ensure _all_ possible types have a `code` property (and any other properties we try to access)
 - narrow the type around attribute access to "prove" to typescript that it's safe (so an `instanceof` or `typeof` check, or maybe even checking `if (it.code) { ... }`)
 - create a `IStatusCodeable` (very bad name, plz dont use) type that requires a `code` and accept that -- this is the worst of the options in _this_ use case

Reply:
Went with option 2. Thanks for explaining. I have a better understanding, but realize I need to be more careful using `union` in typescript.